### PR TITLE
🏷✨ Update @types/fhir

### PR DESCRIPTION
**Pull request by jhuenges**
### Summary of the issue

Updates `@types/fhir` to the latest version `3.0.2` and improved some typings.

### How to reproduce your bugfix or feature
You can run `npm link` and then you can use the linked library by running `npm link @d4l/js-sdk` in the project where you want to use the SDK.

### Due diligence checklist
- [x] Updated version in package.json if applicable.
- [ ] Added/updated tests.
- [ ] Add documentation
- [ ] If this change affects SDK consumers: communicate changes.

### CODE REVIEW

**Review by Glumli — COMMENTED** [review]
Thanks for driving this forward further. This looks great.

File: src/services/appDataService.ts [hunk 1]
```diff
@@ -106,8 +84,9 @@ const appDataService = {
     return recordService.deleteRecord(ownerId, resourceId);
   },
 
-  fetchAllAppData(ownerId: string): Promise<FetchResponse> {
+  fetchAllAppData(ownerId: string, params: Params = {}): Promise<FetchResponse<AppData>> {
     const parameters = prepareSearchParameters({
+      ...params,
```
Comment (Burtchen) [inline src/services/appDataService.ts:89]: Oh nice. What's the reason we're adding this here now?

**Review by Burtchen — APPROVED** [review]
Great addition, thanks a lot. Just one question.

File: src/services/appDataService.ts (hunk 1 above)
Comment (jhuenges) [reply src/services/appDataService.ts:89]: To pass such options like limit and offset. I am pretty sure we don't need them currently, but I thought why not keep the functions as similar as possible

**Pull request merged**